packages/webapp/pages/notifications.tsx

```diff
@@ -59,28 +60,28 @@ const Notifications = (): ReactElement => {
   const { clearUnreadCount } = useNotificationContext();
   const { isSubscribed } = usePushNotificationContext();
 
-  const { mutateAsync: readNotifications } = useMutation(
-    () => gqlClient.request(READ_NOTIFICATIONS_MUTATION),
-    { onSuccess: clearUnreadCount },
-  );
-  const queryResult = useInfiniteQuery<NotificationsData>(
-    ['notifications'],
-    ({ pageParam }) =>
+  const { mutateAsync: readNotifications } = useMutation({
+    mutationFn: () => gqlClient.request(READ_NOTIFICATIONS_MUTATION),
+    onSuccess: clearUnreadCount,
+  });
+  const queryResult = useInfiniteQuery<NotificationsData>({
+    queryKey: ['notifications'],
+    queryFn: ({ pageParam }) =>
       gqlClient.request(NOTIFICATIONS_QUERY, {
         first: 100,
         after: pageParam,
       }),
-    {
-      getNextPageParam: (lastPage) =>
-        lastPage?.notifications?.pageInfo?.hasNextPage &&
-        lastPage?.notifications?.pageInfo?.endCursor,
-      onSuccess: (data) => {
-        if (hasUnread(data)) {
-          readNotifications();
-        }
-      },
-    },
-  );
+    initialPageParam: '',
+    getNextPageParam: ({ notifications }) =>
+      getNextPageParam(notifications?.pageInfo),
+  });
+
+  useEffect(() => {
+    if (queryResult.data && hasUnread(queryResult.data)) {
```

Comment:
Thinking if we should check `queryResult.isSuccess `, but I guess it makes more sense to just look at whatever data is because then we don't have to make distinctions if its cache or not. Seems more easier to reason than loading state. WDYT since you did few of those now?

Reply:
Hmm in this case we also want to do something further with data anyway, so feels like listening to data is fine here?